nixos/modules/hardware/opengl.nix

```diff
@@ -4,35 +4,90 @@ with lib;
 
 let
 
-  cfg = config.hardware.opengl;
+  cfg = config.hardware.impure-drivers;
+
+  impurePath = pkgs.addDriverRunpath.driverLink;
+  impurePath32 = pkgs.pkgsi686Linux.addDriverRunpath.driverLink;
 
   kernelPackages = config.boot.kernelPackages;
 
   videoDrivers = config.services.xserver.videoDrivers;
 
   package = pkgs.buildEnv {
     name = "opengl-drivers";
-    paths = [ cfg.package ] ++ cfg.extraPackages;
+    paths = cfg.packages;
   };
 
   package32 = pkgs.buildEnv {
     name = "opengl-drivers-32bit";
-    paths = [ cfg.package32 ] ++ cfg.extraPackages32;
+    paths = cfg.packages32;
   };
 
 in
 
 {
 
   imports = [
-    (mkRenamedOptionModule [ "services" "xserver" "vaapiDrivers" ] [ "hardware" "opengl" "extraPackages" ])
+    (mkRenamedOptionModule [ "services" "xserver" "vaapiDrivers" ] [ "hardware" "impure-drivers" "packages" ])
     (mkRemovedOptionModule [ "hardware" "opengl" "s3tcSupport" ] "S3TC support is now always enabled in Mesa.")
+
+    # The old option `hardware.opengl.enable` corresponds to `hardware.impure-drivers.opengl.enable` (defaults to `true`).
+    # However, in order support transition of the systems have been setting `hardware.opengl.enable`,
+    # we need to translate it into `hardware.impure-drivers.enable` (defaults to `false`)
+    (mkRenamedOptionModule [ "hardware" "opengl" "enable" ] [ "hardware" "impure-drivers" "enable" ])
```

Comment:
I think the most questionable decision is this. Maybe I shouldn't be introducing `hardware.impure-drivers.opengl.enable` at all?

Reply:
Perhaps leave `hardware.opengl.enable` unchanged and make `hardware.impure-drivers.opengl.enable` depend on its value?

Reply:
H'm, how about I keep `hardware.opengl.enable`, drop `hardware.impure-drivers.opengl`, but add `hardware.impure-drivers.enable = true` to `hardware.opengl`'s `config`?